cron.php

```diff
@@ -0,0 +1,464 @@
+<?php
+/*
+ Plugin Name: WP-Cron Control Revisited
+ Plugin URI:
+ Description: Take control of wp-cron execution.
+ Author: Erick Hitter, Automattic
+ Version: 0.1
+ Text Domain: wp-cron-control-revisited
+ */
+
+// Enable in production only for specific sites
+// Otherwise, class WP-Cron Control is loaded
+$whitelisted_sites = array();
+if ( VIP_GO_ENV && ! in_array( FILES_CLIENT_SITE_ID, $whitelisted_sites ) ) {
+	return;
+}
+
+class WP_Cron_Control_Revisited {
+	/**
+	 * Class instance
+	 */
+	private static $__instance = null;
+
+	public static function instance() {
+		if ( ! is_a( self::$__instance, __CLASS__ ) ) {
+			self::$__instance = new self;
+		}
+
+		return self::$__instance;
+	}
+
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Class properties
+	 */
+	private $namespace = 'wp-cron-control-revisited/v1';
+	private $secret    = null;
+
+	private $job_queue_size                  = 10;
+	private $job_queue_window_in_seconds     = 60;
+	private $job_execution_buffer_in_seconds = 15;
+	private $job_timeout_in_minutes          = 10;
+	private $job_concurrency_limit           = 10;
+
+	private $internal_jobs           = array();
+	private $internal_jobs_schedules = array();
+
+	private $cache_key_lock           = 'wpccr_lock';
+	private $cache_key_lock_timestamp = 'wpccr_lock_ts';
+
+	/**
+	 * Register hooks
+	 */
+	private function __construct() {
+		// For now, leave WP-CLI alone
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return;
+		}
+
+		// Load plugin functionality, when conditions are met
+		if ( defined( 'WP_CRON_CONTROL_SECRET' ) ) {
+			// Block normal cron execution
+			define( 'DISABLE_WP_CRON', true );
+			define( 'ALTERNATE_WP_CRON', false );
+
+			add_action( 'muplugins_loaded', array( $this, 'block_direct_cron' ) );
+			remove_action( 'init', 'wp_cron' );
+
+			add_filter( 'cron_request', array( $this, 'block_spawn_cron' ) );
+
+			// Core plugin functionality
+			$this->prepare();
+			add_action( 'rest_api_init', array( $this, 'rest_api_init' ) );
+			add_filter( 'cron_schedules', array( $this, 'register_internal_events_schedules' ) );
+			add_action( 'muplugins_loaded', array( $this, 'schedule_internal_events' ), 11 );
```

Comment:
My choice of action, and use of priority, was pretty silly in this case; there's no need for it, so I'll find a more-appropriate hook.